### DISCUSSION TITLE
Customize colors of react library in consumer app

### DISCUSSION THREAD

**Discussion by peter-loftus-fbpm** [tailwindlabs/tailwindcss#10013] [Help] [answered]
I'm trying to build a react component library that uses tailwind for styling. This library would have abstract colors that should be overwritten by the consumer application to allow for customization of the components. The library is using Rollup with PostCSS to package and process everything for distribution. Currently the settings in the consumer app doesn't affect the library components when being used in it. Is there some configuration I'm missing? Is this an issue with my rollup or PostCSS config?

I found [this](https://stackoverflow.com/questions/71695814/tailwind-not-being-applied-to-library) article on stack overflow but didn't think it was the correct answer since extending the color theme adds more classes than just `bg-primary`. Is there something I'm missing here?

`tailwind.config.js` from library

```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  darkMode: "class",
  content: ["./src/**/*.{html,js,ts,tsx}"],
  theme: {
    extend: {
      colors: {
        primary: {
          50: "hsl(209, 52%, 95%)",
          100: "hsl(209, 52%, 90%)",
          200: "hsl(209, 52%, 80%)",
          300: "hsl(209, 52%, 70%)",
          400: "hsl(209, 52%, 60%)",
          500: "hsl(209, 52%, 50%)",
          600: "hsl(209, 52%, 40%)",
          700: "hsl(209, 52%, 30%)",
          800: "hsl(209, 52%, 20%)",
          900: "hsl(209, 52%, 10%)",
        },
      },
    },
  },
  plugins: [],
};
```

`tailwind.config.js` from consumer app

```js
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: ["./src/**/*.{js,jsx,ts,tsx}"],
  theme: {
    extend: {
      colors: {
        primary: {
          50: "hsl(45, 100%, 95%)",
          100: "hsl(45, 100%, 90%)",
          200: "hsl(45, 100%, 80%)",
          300: "hsl(45, 100%, 70%)",
          400: "hsl(45, 100%, 60%)",
          500: "hsl(45, 100%, 50%)",
          600: "hsl(45, 100%, 40%)",
          700: "hsl(45, 100%, 30%)",
          800: "hsl(45, 100%, 20%)",
          900: "hsl(45, 100%, 10%)",
        },
      },
    },
  },
  plugins: [],
};
```

**Comment by wongjn** [comment, marked as answer]
If the library package builds the CSS and the consumer consumes that CSS, then the consumer's Tailwind configuration would have no effect on the library CSS, since it is already built. The consumer would need to build the CSS, and include the library source code in its `content` file globs.

**Reply by peter-loftus-fbpm** [reply]
Thank you very much, I've removed the CSS processing when packaging the library and updated the `content` property in the `tailwind.config.js` from the consumer app to include the source from `node_modules` which works perfectly.
